agent/src/beerocks/slave/son_slave_thread.cpp

```diff
@@ -2201,6 +2201,28 @@ bool slave_thread::handle_cmdu_ap_manager_message(
         send_cmdu_to_controller(cmdu_tx);
         break;
     }
+    case beerocks_message::ACTION_APMANAGER_SUPPORTED_CHANNELS_RESPONSE: {
+        LOG(TRACE) << "received ACTION_APMANAGER_SUPPORTED_CHANNELS_RESPONSE";
+        auto response =
+            cmdu_rx.addClass<beerocks_message::cACTION_APMANAGER_SUPPORTED_CHANNELS_RESPONSE>();
+        if (!response) {
+            LOG(ERROR) << "addClass cACTION_APMANAGER_SUPPORTED_CHANNELS_RESPONSE failed";
+            return false;
+        }
+
+        auto tuple_supported_channels = response->supported_channels_list(0);
+        std::copy_n(&std::get<1>(tuple_supported_channels), message::SUPPORTED_CHANNELS_LENGTH,
+                    hostap_params.supported_channels);
```

Comment:
I think we could avoid the additional state by including the supported channels list in ACTION_APMANAGER_JOINED_NOTIFICATION. I think that for now, we can just do the read_supported_channels in the else branch of the acs_enabled condition. In either case, the supported_channels_list will be available.

And actually, the acs_enabled condition should (eventually) be removed entirely and it should be replaced by the preference query that you added.

Reply:
Removed and added read_supported_channels() in else.